cgp/ea/mu_plus_lambda.py

```diff
@@ -108,10 +110,25 @@ def step(
         # population instead of the other way around
         combined = offsprings + pop.parents
 
-        for ind in combined:
-            self.local_search(ind)
+        # we follow a two-step process for selection of new parents:
+        # we first determine the fitness for all individuals, then, if
+        # applicable, we apply local search to the k_local_search
+        # fittest individuals; after this we need to recompute the
+        # fitness for all individuals for which parameters changed
+        # during local search; finally we sort again by fitness, now
+        # taking into account the effect of local search for
+        # subsequent selection
         combined = self._compute_fitness(combined, objective)
+        combined = self._sort(combined)
+
+        if self.k_local_search is None:
+            k_local_search = len(combined)
+        else:
+            k_local_search = self.k_local_search
```

Comment:
what about putting this in one line? :-)

Reply:
ok, in this case it's nice :P